lib/conduit_async_ssl.mli

```diff
@@ -33,7 +33,7 @@ val ssl_connect :
   ?verify:(Ssl.Connection.t -> bool Deferred.t) ->
   Reader.t ->
   Writer.t ->
-  (Reader.t * Writer.t) Deferred.t
+  (Reader.t * Writer.t) Deferred.Or_error.t
```

Comment:
One thing that the `Or_error` contract is supposed to guarantee is that these functions shouldn't raise. Are we certain this isn't the case?